### ISSUE TITLE
lint: add "onlyExported" option for unnamedResult checker

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#345]
One of the main reasons for named results is godoc and API clarity.
For unexported things this may be a little too opinionated for many developers.
We should provide a switch that limits this check to exported functions/methods only.
Don't know what value fits default better. Probably `false`, to force more named results by default.

**Comment by Quasilyte** [comment]
Some signatures from upspin project that trigger this check:
```
func (c *cache) cacheName(de *upspin.DirEntry) (string, string)
func (s *web) accessAll(name upspin.PathName) (bool, bool, error)
func (i *Iterator) GetAndAdvance() (interface{}, interface{}, bool)
```
They could be like this:
```go
func (c *cache) cacheName(de *upspin.DirEntry) (dir string, file string)
func (s *web) accessAll(name upspin.PathName) (readable bool, listable bool, err error)
func (i *Iterator) GetAndAdvance() (key interface{}, value interface{}, valie bool)
```

**Comment by cristaloleg** [comment]
Closed #446
